src/main/scala/org/ergoplatform/Input.scala

```diff
@@ -21,6 +21,8 @@ case class DataInput(boxId: BoxId) {
     case x: DataInput => util.Arrays.equals(boxId, x.boxId)
     case _ => false
   }
+
+  override def hashCode(): Int = boxId.hashCode
```

Comment:
Need to use deepHashCode, since boxId is Array

Reply:
Whoa, that was silly. Thanks! Fixed.